### DISCUSSION TITLE
updateListingProperty (v3) only takes the last value (even when providing multiple values) - python code

### DISCUSSION THREAD

**Discussion by go4itrs** [etsy/open-api#1086] [Q&A]
Trying to create/update the listing attribute: "Room". (Which is a simple multi-select drop-down with values like "Bedroom", "Bathroom", "Kids", "Nursery" etc.

When calling the v3 api, only the last value of the value_ids is selected, what am I doing wrong?

--
Python code:
--
 url = f"https://openapi.etsy.com/v3/application/shops/{v_shop_id}/listings/{v_listing_id}/properties/{v_property_id}"

    headers = {
        "Accept": "application/x-www-form-urlencoded",
        "x-api-key": v_keystring,
        "Authorization": f"Bearer {v_token}"
    }

    data = {
            'value_ids': [2354,2356],
            'values':['Bedroom','Bathroom']
    }

    response = requests.put(url=url, headers=headers, data=data)

--
after running the code only the "Bathroom" value is checked (was expecting both Bedroom and Bathroom)

When I change the order,  eg. 'values':['Bathroom', 'Bedroom'] and its corresponding value_ids, only the "Bedroom" is checked in the drop down.
No error messages.
Any pointers to get this fixed?
Go4it, Rene

**Comment by fifdee** [comment]
Same problem here.
I also tried putting each value_id - value pair in separate request, but the output was the same, only last value selected (even though each request didn't throw any errors).

**Comment by slider1234** [comment]
`import requests

url = f"https://openapi.etsy.com/v3/application/shops/{v_shop_id}/listings/{v_listing_id}/properties/{v_property_id}"

headers = {
    "x-api-key": v_keystring,
    "Authorization": f"Bearer {v_token}",
    "Content-Type": "application/x-www-form-urlencoded"
}

data = {
    'value_ids': '2354,2356',  # Comma-separated string of IDs
    'values': 'Bedroom,Bathroom'  # Comma-separated string of values
}

response = requests.put(url=url, headers=headers, data=data)


print(response.status_code)
print(response.text)
`

**Reply by jk1981lv** [reply]
Thank you. This solution works. Completely different that Etsy's  API documentation 😀
